internal/services/manageddevopspools/managed_devops_pool_resource.go

```diff
@@ -0,0 +1,726 @@
+)
+
+type ManagedDevOpsPoolResource struct{}
+
+type ManagedDevOpsPoolModel struct {
+	DevCenterProjectResourceId     string                                `tfschema:"dev_center_project_resource_id"`
+	VmssFabricProfile              []VmssFabricProfileModel              `tfschema:"vmss_fabric_profile"`
+	Identity                       []identity.ModelUserAssigned          `tfschema:"identity"`
+	Location                       string                                `tfschema:"location"`
+	MaximumConcurrency             int64                                 `tfschema:"maximum_concurrency"`
+	Name                           string                                `tfschema:"name"`
+	AzureDevOpsOrganizationProfile []AzureDevOpsOrganizationProfileModel `tfschema:"azure_devops_organization_profile"`
+	ResourceGroupName              string                                `tfschema:"resource_group_name"`
+	Tags                           map[string]string                     `tfschema:"tags"`
+	StatefulAgentProfile           []StatefulAgentProfileModel           `tfschema:"stateful_agent_profile"`
+	StatelessAgentProfile          []StatelessAgentProfileModel          `tfschema:"stateless_agent_profile"`
+}
+
+func (ManagedDevOpsPoolResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-.]*[a-zA-Z0-9-]$`),
+				"`name` can only include alphanumeric characters, periods (.) and hyphens (-). It must also start with alphanumeric characters and cannot end with periods (.).",
+			),
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
+		"location":            commonschema.Location(),
+		"azure_devops_organization_profile": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"organization": {
+						Type:     pluginsdk.TypeList,
+						Required: true,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"url": {
+									Type:     pluginsdk.TypeString,
+									Required: true,
+									ValidateFunc: validation.All(
+										validation.IsURLWithHTTPS,
+										validation.StringMatch(
+											regexp.MustCompile(`[a-zA-Z0-9]$`),
+											"url must end with a letter or number",
+										),
+									),
+								},
+								"parallelism": {
+									Type:         pluginsdk.TypeInt,
+									Optional:     true,
+									ValidateFunc: validation.IntBetween(1, 10000),
+								},
+								"projects": {
+									Type:     pluginsdk.TypeList,
+									Optional: true,
+									Elem: &pluginsdk.Schema{
+										Type:         pluginsdk.TypeString,
+										ValidateFunc: validate.ProjectName,
+									},
+								},
+							},
+						},
+					},
+					"permission_profile": {
```

Comment:
Insert a blank line between top-level schema blocks to align with the provider. This formatting should also apply to other schemas.

```suggestion
					},

					"permission_profile": {
```